internal/resource/http.go

```diff
@@ -247,3 +253,21 @@ func (c HttpClient) getReaderWithHeader(url string, header http.Header) (io.Read
 		}
 	}
 }
+
+// addSystemInfoHeaders will add an ID, a VERSION_ID (if present), and a
+// VARIANT_ID (if present) header to the client's ignition config GET request
+func addSystemInfoHeaders(header *http.Header) error {
+	or, err := osrelease.New()
```

Comment:
This is parsing the values from initramfs os-release (i.e. ID=dracut), not the ones from the sysroot (i.e. ID=coreos). Is this the expected behavior?

Reply:
This I think came from my example I shared with @Rahuls0720. I failed to point out that the newer constructor should be used here to use the one in sysroot.

Reply:
Depending on **when** this is called though, the sysroot may not yet be be mounted/available I fear.

Reply:
That possible. In @Rahuls0720's testing we should be able to verify and modify the code as needed.

Reply:
This looks good to me.

Reply:
@lucab is right, sysroot isn't mounted at that point. There are further discussions on this in the #forum-os group on slack.